src/lib/RelayOrderLib.sol

```diff
@@ -45,50 +48,29 @@ library RelayOrderLib {
         }
     }
 
-    function toPermit(RelayOrder memory order)
+    function transferInputTokens(RelayOrder memory order, IPermit2 permit2, bytes calldata sig)
         internal
-        pure
-        returns (ISignatureTransfer.PermitBatchTransferFrom memory permit)
+        returns (ResolvedTransferDetails memory resolved)
     {
-        uint256 inputsLength = order.inputs.length;
-        // Build TokenPermissions array with the maxValue
-        ISignatureTransfer.TokenPermissions[] memory permissions =
-            new ISignatureTransfer.TokenPermissions[](inputsLength);
-
-        for (uint256 i = 0; i < inputsLength; i++) {
-            Input memory input = order.inputs[i];
-            permissions[i] = ISignatureTransfer.TokenPermissions({token: input.token, amount: input.maxAmount});
-        }
-
-        return ISignatureTransfer.PermitBatchTransferFrom({
-            permitted: permissions,
-            nonce: order.info.nonce,
-            deadline: order.info.deadline
-        });
-    }
-
-    /// @notice The requestedAmount is built from the decayed/resolved amount.
-    function toTransferDetails(RelayOrder memory order)
-        internal
-        view
-        returns (ISignatureTransfer.SignatureTransferDetails[] memory details)
-    {
-        uint256 inputsLength = order.inputs.length;
-        // Build TransferDetails with the final resolved amount
-        details = new ISignatureTransfer.SignatureTransferDetails[](inputsLength);
-
-        for (uint256 i = 0; i < inputsLength; i++) {
-            Input memory input = order.inputs[i];
-            address recipient = input.recipient == address(0) ? msg.sender : input.recipient;
-            details[i] = ISignatureTransfer.SignatureTransferDetails({
-                to: recipient,
-                requestedAmount: RelayDecayLib.decay(
-                    input.startAmount, input.maxAmount, order.decayStartTime, order.decayEndTime
-                    )
-            });
-        }
-
-        return details;
+        bytes32 orderHash = order.hash();
+        (
+            ISignatureTransfer.TokenPermissions[] memory permissions,
+            ISignatureTransfer.SignatureTransferDetails[] memory details
+        ) = order.inputs.toPermitDetails(order.decayStartTime, order.decayEndTime);
+
+        permit2.permitWitnessTransferFrom(
+            ISignatureTransfer.PermitBatchTransferFrom({
+                permitted: permissions,
+                nonce: order.info.nonce,
+                deadline: order.info.deadline
+            }),
+            details,
+            order.info.swapper,
+            orderHash,
+            RelayOrderLib.PERMIT2_ORDER_TYPE,
+            sig
+        );
+        return ResolvedTransferDetails({swapper: order.info.swapper, transferDetails: details, orderHash: orderHash});
```

Comment:
whats the point of returning this?

Reply:
We need to get the "resolved" amounts somehow in the quoter.. couldn't think of a better way since we dont have the callback functionality like we do in X. But lmk if I'm missing something

Reply:
Talked offline but iirc we can use the offchain quoter to resolve the input amounts, I think the only new information that this would return would be what `feeRecipient` the filler passed in when executing the order?

Reply:
Yea only reason I can see of returning the resolved amount on chain is for onchain integrators? I.e. using the tip amount as input for something else

Reply:
yea i can take it out.. was just trying to emulate UniswapX quoter but because we don't have callbacks we can't get any info.. if we dont need this info though let's just take it out ?